salt/modules/saltutil.py

```diff
@@ -378,28 +378,31 @@ def refresh_grains(**kwargs):
         The available execution modules will be reloaded as part of this
         proceess, as grains can affect which modules are available.
 
-    refresh_pillar : True
-        Set to ``False`` to keep pillar data from being refreshed.
+    :param wait:            Wait for pillar refresh to complete, defaults to ``False``.
+    :type wait:             bool, optional
+    :param timeout:         How long to wait in seconds, only used when wait is True, defaults to 30.
+    :type timeout:          int, optional
+    :param refresh_pillar:  Set to ``False`` to keep pillar data from being refreshed
+    :type refresh_pillar:   bool, optional
+    :return:                Boolean status, True when the refresh_grains event was fired successfully.
 
     CLI Examples:
 
     .. code-block:: bash
 
         salt '*' saltutil.refresh_grains
+        salt '*' saltutil.refresh_grains wait=true timeout=30 refresh_pillar=False
     """
-    kwargs = salt.utils.args.clean_kwargs(**kwargs)
-    _refresh_pillar = kwargs.pop("refresh_pillar", True)
-    if kwargs:
-        salt.utils.args.invalid_kwargs(kwargs)
     # Modules and pillar need to be refreshed in case grains changes affected
     # them, and the module refresh process reloads the grains and assigns the
     # newly-reloaded grains to each execution module's __grains__ dunder.
-    if _refresh_pillar:
-        # we don't need to call refresh_modules here because it's done by refresh_pillar
```

Comment:
I think that statement was a lie, I checked the refresh_pillar function and it didn't do refresh_modules. 
Please correct me if I'm wrong.